### ISSUE TITLE
[BUG] BlobContainerClient readAddress(..) failed: Connection reset by peer

### ISSUE THREAD

**Issue by rvvaeke** [Azure/azure-sdk-for-java#23532]
**Describe the bug**
When starting a Spring Boot application using the azure-spring-boot-starter-storage library, an error occurs when interacting with the BlobContainerClient through the exists() method.

***Exception or Stack Trace***
```
2021-08-12 11:18:26,894 39925 [my-calendar-service,,,][parallel-1] INFO  i.n.h.s.ReferenceCountedOpenSslContext - BoringSSL doesn't allow to enable or disable TLSv1.3 ciphers explicitly. Provided TLSv1.3 ciphers: 'TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384', default TLSv1.3 ciphers that will be used: 'TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256'.
2021-08-12 11:18:28,100 41131 [my-calendar-service,,,][reactor-http-epoll-2] WARN  r.n.http.client.HttpClientConnect - [id:9d445f68, L:/10.253.132.29:57588 - R:cld2paaswgsskalbeheerd.blob.core.windows.net/52.169.240.64:443] The connection observed an error
io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
2021-08-12 11:18:28,148 41179 [my-calendar-service,,,][reactor-http-epoll-2] WARN  r.n.http.client.HttpClientConnect - [id:9d445f68, L:/10.253.132.29:57588 ! R:cld2paaswgsskalbeheerd.blob.core.windows.net/52.169.240.64:443] The connection observed an error
java.nio.channels.ClosedChannelException: null
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1066)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:831)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

... 

Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [be.liantis.lig.service.kalender.repository.azure.AzureBlobResourceRepository]: Factory method 'rapportResourceRepository' threw exception; nested exception is reactor.core.Exceptions$ReactiveException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
	... 219 common frames omitted
Caused by: reactor.core.Exceptions$ReactiveException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at reactor.core.Exceptions.propagate(Exceptions.java:392)
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:97)
	at reactor.core.publisher.Mono.block(Mono.java:1704)
	at com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout(StorageImplUtils.java:128)
	at com.azure.storage.blob.BlobContainerClient.existsWithResponse(BlobContainerClient.java:228)
	at com.azure.storage.blob.BlobContainerClient.exists(BlobContainerClient.java:211)
	at be.liantis.lig.service.kalender.repository.azure.AzureBlobResourceRepository.<init>(AzureBlobResourceRepository.java:38)
	at be.liantis.lig.service.kalender.config.resource.ResourceRepositoryConfiguration$AzureStorageBlobResourceRepositoryConfiguration.rapportResourceRepository(ResourceRepositoryConfiguration.java:31)
	at be.liantis.lig.service.kalender.config.resource.ResourceRepositoryConfiguration$AzureStorageBlobResourceRepositoryConfiguration$$EnhancerBySpringCGLIB$$87bdc2e4.CGLIB$rapportResourceRepository$0(<generated>)
	at be.liantis.lig.service.kalender.config.resource.ResourceRepositoryConfiguration$AzureStorageBlobResourceRepositoryConfiguration$$EnhancerBySpringCGLIB$$87bdc2e4$$FastClassBySpringCGLIB$$547761b7.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331)
	at be.liantis.lig.service.kalender.config.resource.ResourceRepositoryConfiguration$AzureStorageBlobResourceRepositoryConfiguration$$EnhancerBySpringCGLIB$$87bdc2e4.rapportResourceRepository(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 220 common frames omitted
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99)
		... 236 common frames omitted
```

**To Reproduce**
1. Start Spring Boot app
2. Interact with a BlobContainerClient on Spring Context startup

***Code Snippet***
```
public AzureBlobResourceRepository(@NonNull BlobServiceClient blobServiceClient, @NonNull String containerName, @NonNull AccessTier accessTier) {
        this.containerName = containerName;
        this.accessTier = accessTier;
        this.azureStorageResourcePatternResolver = new AzureStorageResourcePatternResolver(blobServiceClient);
        this.blobContainerClient = blobServiceClient.getBlobContainerClient(containerName);
        if (!this.blobContainerClient.exists()) {
            this.blobContainerClient.create();
        }
    }
```

**Expected behavior**
The requests succeeds.

**Setup (please complete the following information):**
 - OS: amd64 Linux 3.10.0-1160.31.1.el7.x86_64
 - IDE: IntelliJ 2021.1.1
 - Library/Libraries: 
 - - org.springframework.boot:spring-boot-starter-parent:2.5.3
 - - com.azure.spring:azure-spring-boot-starter-storage:3.7.0
 - - io.netty:*:4.1.66.Final
 - - io.projectreactor.netty:*:1.0.9
 - Java version: 8
 - App Server/Environment: Embedded Tomcat 9.0.50
 - Frameworks: Spring Boot

**Additional context**
Already checked with network team to verify the issues is not firewall related. Below snippet proofs communication from the server to Azure Storage works:
```
lnxdcom@dcrx-las-d005 ~]$ telnet cld2paaswgsskalbeheerd.blob.core.windows.net 443
Trying 52.169.240.64...
Connected to cld2paaswgsskalbeheerd.blob.core.windows.net.
```

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x ] Bug Description Added
- [x ] Repro Steps Added
- [x ] Setup information Added

**Comment by rickle-msft** [comment]
@stliu Can you take a look at this please?

**Comment by rickle-msft** [comment]
@rvvaeke It's possible the version of the Storage sdk you are using is out of date. Could you try upgrading to latest?

**Comment by rvvaeke** [comment]
@rickle-msft we are using the latest version of azure-spring-boot-starter-storage:3.7.0, which uses azure-storage-blob:12.12.0. I noticed the latest version of azure-storage-blob is 12.13.0, so I tried with that version. Upgrading to com.azure:azure-storage-blob:12.13.0 in fact solved the problem. 

I look forward to seeing a new version of azure-spring-boot-starter-storage which uses com.azure:azure-storage-blob:12.13.0.
